hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -966,6 +967,126 @@ int SaraNcpClient::checkNetConfForImsi() {
+                } else {
+                    newProf = static_cast<int>(UbloxSaraUmnoprof::STANDARD_EUROPE);
+                }
+            }
+            // KORE AT&T or 3rd Party SIM
+            else {
+                // break out of do-while if we're trying to set SIM_SELECT a third time
+                if (resetCount >= 2) {
+                    LOG(WARN, "UMNOPROF=1 did not resolve a built-in profile, please check if UMNOPROF=100 is required!");
+                    break;
+                }
+            }
+
+            // Disconnect before making changes to the UMNOPROF
+            auto respCfun = parser_.sendCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN?");
+            int cfunVal = -1;
+            auto retCfun = CHECK_PARSER(respCfun.scanf("+CFUN: %d", &cfunVal));
+            CHECK_PARSER_OK(respCfun.readResult());
+            if (retCfun == 1 && cfunVal != 0) {
+                CHECK_PARSER_OK(parser_.execCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN=0,0"));
+            }
+            // This is a persistent setting
+            parser_.execCommand(1000, "AT+UMNOPROF=%d", newProf);
+            // Not checking for error since we will reset either way
+            reset = true;
+            disableLowPowerModes = true;
+        } else if (r == 1 && static_cast<UbloxSaraUmnoprof>(curProf) == UbloxSaraUmnoprof::STANDARD_EUROPE) {
+            auto respBand = parser_.sendCommand(UBLOX_UBANDMASK_TIMEOUT, "AT+UBANDMASK?");
+            uint64_t ubandUint64 = 0;
+            char ubandStr[24] = {};
+            auto retBand = CHECK_PARSER(respBand.scanf("+UBANDMASK: 0,%23[^,]", ubandStr));
+            CHECK_PARSER_OK(respBand.readResult());
+            if (netConf_.netProv() == CellularNetworkProvider::TWILIO && retBand == 1) {
+                char* pEnd = &ubandStr[0];
+                ubandUint64 = strtoull(ubandStr, &pEnd, 10);
+                // Only update if Twilio Super SIM and not set to correct bands
+                if (pEnd - ubandStr > 0 && ubandUint64 != 6170) {
+                    // Enable Cat-M1 bands 2,4,5,12 (AT&T), 13 (VZW) = 6170
+                    parser_.execCommand(UBLOX_UBANDMASK_TIMEOUT, "AT+UBANDMASK=0,6170");
+                    // Not checking for error since we will reset either way
+                    reset = true;
+                    disableLowPowerModes = false;
+                }
+            }
+        }
+        if (reset) {
+            CHECK_PARSER_OK(parser_.execCommand("AT+CFUN=15,0"));
+            HAL_Delay_Milliseconds(2000);
+
+            CHECK(waitAtResponseFromPowerOn(state));
+
+            // Checking for SIM readiness ensures that other related commands
+            // like IFC or PSM/eDRX won't error out
+            CHECK(checkSimReadiness());
+            // Prevent modem from immediately dropping into PSM/eDRX modes
+            // which (on 05.12) may be enabled as soon as the UMNOPROF has taken effect
+            if (disableLowPowerModes) {
+                // Not checking the error
```

Comment:
Any specific reason? If something goes wrong when we try to disable PSM/eDRX we may accidentally continue initialization with either of those enabled.

Reply:
I have seen some runs fail PSM with something like the following
```
AT+CPSMS=0
+CME ERROR: SIM Failure
```
, and I thought not to make this a blocking call. But then, we are looking for `CHECK(checkSimReadiness())` so it should be OK. I will add that in.